lib/capdrupal.rb

```diff
@@ -0,0 +1,310 @@
+            warn "backup file #{backup} does not exist."
+            next ""
+          end
+
+          # Revert from backup.
+          execute :drush, "sql:cli < #{backup}"
+
+          # Delete the unziped backup.
+          execute :rm, "#{backup}"
+        end
+      end
+    end
+
+    desc "Backup drupal site database"
+    task :backup do
+      on roles(:app) do
+        within release_path.join(fetch(:app_path)) do
+          backup_destination = "#{shared_path}/#{fetch(:backup_path)}/#{fetch(:stage)}_#{release_timestamp}"
+          execute :drush, "sql:dump --result-file=#{backup_destination} --gzip"
+        end
+      end
+    end
+
+    namespace :backup do
+      desc "Check required files and directories exist"
+      task :check do
+        on release_roles :all do
+          execute :mkdir, "-p", "#{shared_path}/#{fetch(:backup_path)}"
+        end
+      end
+
+      desc "Clean up old drupal site database backup"
+      task :cleanup do
+        on roles(:app) do
+          within release_path.join(fetch(:app_path)) do
+            backup_path = "#{shared_path}/#{fetch(:backup_path)}"
+            keep_backups = fetch(:keep_backups, 5)
+
+            # Fetch every file from the backup directory, oldest on top.
+            backups = capture(:ls, "-ltrx", backup_path).split
+            info "Keeping #{keep_backups} of #{backups.count} backup dump on #{backup_path}."
+
+            # If we found less file than the keep number finish the task here
+            next "" unless backups.count > keep_backups.to_i
+
+            # Calculate number of file to delete.
+            to_delete = backups.count - keep_backups.to_i
+
+            # Loop throught file to delete (oldest files on top).
+            backups[0, to_delete].map do |backup|
+              execute :rm, "#{backup_path}/#{backup}"
+            end
+          end
+        end
+      end
+    end
+  end
+
+  namespace :cache do
+    desc 'Clear all caches'
+    task :clear do
+      on roles(:app) do
+        within release_path.join(fetch(:app_path)) do
+          execute :drush, 'cr'
+        end
+      end
+    end
+  end
+
+  namespace :module do
+    desc 'Enable module(s)'
+    task :enable do
+      on roles(:app) do
+        within release_path.join(fetch(:app_path)) do
+          for mod in fetch(:enable_modules)
+            execute :drush, "en '#{mod}' -y"
+          end
+        end
+      end
+    end
+
+    desc 'Disable module(s)'
+    task :disable do
+      on roles(:app) do
+        within release_path.join(fetch(:app_path)) do
+          for mod in fetch(:disable_modules)
+            execute :drush, "pmu '#{mod}' -y"
+          end
+        end
+      end
+    end
+  end
+
+  namespace :maintenance do
+    desc "Set maintenance mode"
+    task :on do
+      on roles(:app) do
+        within release_path.join(fetch(:app_path)) do
+          execute :drush, "state:set system.maintenance_mode 1 -y"
+          execute :drush, 'cr'
```

Comment:
```suggestion
          execute :drush, 'cr', raise_on_non_zero_exit: false
```

In some situation this is really too early to run a cache clear (like having hard dependencies between two modules but one that is not already enabled). I propose to let this command failed on these situation.

Reply:
@yannlugrin @WengerK do you agree with my suggestion ?

Reply:
agree